File: frontend/src/types/task.ts
export type TaskPriority = "low" | "medium" | "high";
export type TaskStatus = "pending" | "in_progress" | "finished";

export type Task = {
  id: string;
  title: string;
  description: string;
  subject: string;
  due_date: string;
  priority: TaskPriority;
  status: TaskStatus;
  created_at: string;
  updated_at: string;
};

export type TaskPayload = Omit<Task, "id" | "created_at" | "updated_at">;

export type Summary = {
  total: number;
  pending: number;
  finished: number;
  highPriority: number;
};

export type Filters = {
  status?: TaskStatus | "";
  priority?: TaskPriority | "";
  subject?: string;
};

File: backend/app/schemas/task.py
from pydantic import BaseModel
from typing import Literal

TaskPriority = Literal["low", "medium", "high"]
TaskStatus = Literal["pending", "in_progress", "finished"]


class TaskBase(BaseModel):
    title: str
    description: str
    subject: str
    due_date: str
    priority: TaskPriority
    status: TaskStatus


class TaskCreate(TaskBase):
    pass


class TaskUpdate(TaskBase):
    pass


class TaskOut(TaskBase):
    id: str
    created_at: str
    updated_at: str
